app/admin/cp_benchmarks.rb

```diff
@@ -0,0 +1,58 @@
+ActiveAdmin.register CP::Benchmark do
+  config.sort_order = 'release_date_desc'
+
+  menu parent: 'TPI', priority: 3, label: 'Carbon Performance Benchmarks'
+
+  permit_params :scenario, :sector_id, :release_date, :emissions
+
+  filter :release_date
+  filter :sector
+
+  data_export_sidebar 'CPBenchmarks'
+
+  show do
+    attributes_table do
+      row :id
+      row :release_date
+      row :sector
+      row :scenario
+      row :created_at
+      row :updated_at
+    end
+
+    panel 'Benchmark emission values' do
+      render 'admin/cp/emissions_table', emissions: resource.emissions
+    end
+  end
+
+  csv do
+    year_columns = collection.flat_map(&:emissions_all_years).uniq.sort
```

Comment:
It might be more readable to extract this somewhere. Maybe some decorator / helper?

Reply:
Hmm, just this one line? not sure if that would help for me this is readable